internal/core/application/market_price_service.go

```diff
@@ -162,14 +176,28 @@ func (m *marketPriceService) GetPrices(
 				quotePriceInRefCurrency = q
 			}
 
+			var averageReferentPrice decimal.Decimal
+			quoteAssetTicker, err := m.raterSvc.GetAssetCurrency(v1.QuoteAsset)
+			if err == nil {
+				unitOfQuotePriceInRefCurrency, _ := m.raterSvc.ConvertCurrency(
+					ctx,
+					quoteAssetTicker,
+					referenceCurrency,
+				)
+
+				averageReferentPrice = vwapPerMarket[k].Mul(unitOfQuotePriceInRefCurrency)
```

Comment:
The average price is a single value that takes into account all markets from all different providers.
ATM this would be the average price for each single market which doesn't make that much sense.